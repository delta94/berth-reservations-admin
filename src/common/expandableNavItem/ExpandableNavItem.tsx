import React, { useState } from 'react';
import classNames from 'classnames';
import { IconAngleDown } from 'hds-react';

import styles from './expandableNavItem.module.scss';

export interface ExpandableProps {
  children: React.ReactNode;
  icon?: JSX.Element;
  label: React.ReactNode;
  onClick?: Function;
}

const ExpandableNavItem = ({ label, onClick, children, icon }: ExpandableProps) => {
  const [expanded, setExpanded] = useState(false);
  const handleClick = () => {
    setExpanded(!expanded);
    onClick && onClick();
  };

  return (
    <>
      <div className={styles.expandableNavItem} onClick={handleClick}>
        <div className={classNames(styles.headerBtn, styles.label)} role="button">
          {icon && <div className={styles.icon}>{icon}</div>}
          <div className={styles.label}>{label}</div>
        </div>
        {children && (
          <div
            className={classNames(styles.headerBtn, styles.arrow, {
              [styles.up]: expanded,
            })}
          >
            <IconAngleDown />
          </div>
        )}
      </div>
      {children && (
        <div
          className={classNames(styles.children, {
            [styles.expanded]: expanded,
          })}
        >
          {children}
        </div>
      )}
    </>
  );
};

export default ExpandableNavItem;
