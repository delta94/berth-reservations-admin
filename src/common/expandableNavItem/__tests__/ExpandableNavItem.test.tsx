import React from 'react';
import { shallow } from 'enzyme';

import ExpandableNavItem, { ExpandableProps } from '../ExpandableNavItem';

describe('ExpandableNavItem', () => {
  const onClick = jest.fn();

  beforeEach(() => {
    jest.restoreAllMocks();
  });

  const getWrapper = (props?: ExpandableProps) =>
    shallow(
      <ExpandableNavItem label="Click me" onClick={onClick} {...props}>
        <div>one</div>
        <div>two</div>
      </ExpandableNavItem>
    );

  it('renders normally', () => {
    expect(getWrapper().render()).toMatchSnapshot();
  });

  it('calls the supplied onClick when the item is clicked', () => {
    const item = getWrapper().find('.expandableNavItem');
    item.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
