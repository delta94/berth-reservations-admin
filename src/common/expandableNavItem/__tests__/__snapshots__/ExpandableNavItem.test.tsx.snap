// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableNavItem renders normally 1`] = `
Array [
  <div
    class="expandableNavItem"
  >
    <div
      class="headerBtn label"
      role="button"
    >
      <div
        class="label"
      >
        Click me
      </div>
    </div>
    <div
      class="headerBtn arrow"
    >
      <svg
        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M0 0h24v24H0z"
          />
          <path
            d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
  </div>,
  <div
    class="children"
  >
    <div>
      one
    </div>
    <div>
      two
    </div>
  </div>,
]
`;
